File: 0258-add-digits/0258-add-digits.py
class Solution:
    def addDigits(self, num: int) -> int:
        
        
        while(num):
            c=0
            while(num):
                n=num%10
                c+=n
                num//=10
            if(c//10!=0):
                num=c
            else:
                return c
        return num

        

File: 0258-add-digits/0258-add-digits.cpp
class Solution {
public:
    int addDigits(int num) {
        int k;
        while(num){
            int z=0;
            while(num){
                k=num%10;
                z+=k;
                num/=10;    
            }
            if(z/10!=0){
                num=z;
            }
            else{
                return z;
            }
        }
        return num;
        
    }
};